inject-java/src/main/java/io/micronaut/annotation/processing/visitor/JavaClassElement.java

```diff
@@ -211,37 +211,31 @@ protected void accept(DeclaredType declaringType, javax.lang.model.element.Eleme
                         getterReturnType = mirrorToClassElement(returnType, visitorContext, JavaClassElement.this.genericTypeInfo);
                     }
 
-                    if (getterReturnType != null) {
```

Comment:
Removing this check doesn't seem correct. The previous calls `mirrorToClassElement` could have returned null